### ISSUE TITLE
Question about Calico IPPool creation 

### ISSUE THREAD

**Issue by yboaron** [projectcalico/calico#7845]
To make Calico compatible with Submariner, it needs to create Calico IP pools for the subnets associated with the Pod and Service CIDRs of the remote clusters[1].

We want to make it easier to deploy Submariner on clusters with Calico by automating IP pools creation needed for Submariner functionality and since Submariner has all remote clusters details we thought to automate it in Submariner. 

IIUC, we should use projectcalico.org/v3 to create IP pools and not crd.projectcalico.org/v1,
I have this WIP PR (see [2]) that uses Calico API to create the pools but it doesn't work if Calico API server is not running on the cluster.

What's the right way to automate IP pools creation ? assuming that Calico API server is not running on some clusters

Thanks 
Yossi

[1] 
https://submariner.io/operations/deployment/calico/
[2] https://github.com/submariner-io/submariner/pull/2576/files#diff-fd1083c65cac995ac1b4c41f2d76315763426e26a579c0fd963c752ead6cfb7bR166

**Comment by sridhartigera** [comment]
@yboaron We have been thinking about this and hopefully we can have something working soon.

**Comment by caseydavenport** [comment]
Yeah, right now if the Calico API server isn't running the only surefire way is to use calicoctl, which performs the necessary validation / defaulting and extra logic client-side. 

You could look at what calicoctl is doing if you wanted to try to replicate it yourself, but I couldn't truly recommend that as a long-term solution. 

Like @sridhartigera there's some thought going into this - we know it's a rough edge and are investigating ways to improve it.

**Comment by yboaron** [comment]
Yeah, we would prefer to use the new API as soon as it is available, 
I assume that meanwhile, we'll document that running Calico API server is a prerequisite for automating IPPools creation by Submariner.

Thanks @caseydavenport @sridhartigera

**Comment by yboaron** [comment]
@sridhartigera , any update on this ? 
10x
